File: 1/aoc1.py
def find_two_numbers_totaling(val, lst):
    for i in lst:
        diff = val - i
        if diff in lst:
            return i * diff


def find_three_numbers_totaling(val, lst):
    for i in lst:
        diff = val - i
        match = find_two_numbers_totaling(diff, nums)
        if match:
            return i * match


with open('expenses.txt') as f:
    nums = [int(i) for i in f]
    print('Part 1:', find_two_numbers_totaling(2020, nums))
    print('Part 2:', find_three_numbers_totaling(2020, nums))

File: 1/aoc1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func contains(s []int, e int) bool {
	for _, a := range s {
		if a == e {
			return true
		}
	}
	return false
}

func multiply_two_numbers_totaling(val int, list []int) int {
	for _, v := range list {
		diff := val - v
		if contains(list, diff) {
			return diff * v
		}
	}
	return 0
}

func multiply_three_numbers_totaling(val int, list []int) int {
	for _, v := range list {
		diff := val - v
		match := multiply_two_numbers_totaling(diff, list)
		if match != 0 {
			return v * match
		}
	}
	return 0
}

func main() {
	f, _ := os.Open("expenses.txt")
	defer f.Close()

	var list []int
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		val, _ := strconv.Atoi(scanner.Text())
		list = append(list, val)
	}
	fmt.Println(multiply_two_numbers_totaling(2020, list))
	fmt.Println(multiply_three_numbers_totaling(2020, list))
}
